GDJS/Runtime/variable.js

```diff
@@ -20,278 +26,359 @@
+    : this._type === 'array'
+    ? this._childrenList
+    : [];
+};
+
+/**
+ * Return the length of the
+ * @return {number}
+ */
+gdjs.Variable.prototype.getChildrenCount = function () {
+  if (this.isPrimitive()) return 0;
+  //@ts-ignore This cannot return null as type isn't a primitive.
+  return this.getAllChildrenList().length;
 };
 
 /**
  * Add the given number to the variable value
- * @method
- * @param {number} val the number to add
+ * @param {number} val
  */
-gdjs.Variable.prototype.add = function(val) {
-	this.setNumber(this.getAsNumber()+val);
+gdjs.Variable.prototype.add = function (val) {
+  this.setNumber(this.getAsNumber() + val);
 };
 
 /**
  * Subtract the given number to the variable value
- * @method
- * @param {number} val the number to subtract
+ * @param {number} val
  */
-gdjs.Variable.prototype.sub = function(val) {
-	this.setNumber(this.getAsNumber()-val);
+gdjs.Variable.prototype.sub = function (val) {
+  this.setNumber(this.getAsNumber() - val);
 };
 
 /**
  * Multiply the variable value by the given number
- * @method
- * @param {number} val the factor
+ * @param {number} val
  */
-gdjs.Variable.prototype.mul = function(val) {
-	this.setNumber(this.getAsNumber()*val);
+gdjs.Variable.prototype.mul = function (val) {
+  this.setNumber(this.getAsNumber() * val);
 };
 
 /**
  * Divide the variable value by the given number
- * @method
- * @param {number} val the divisor
+ * @param {number} val
  */
-gdjs.Variable.prototype.div = function(val) {
-	this.setNumber(this.getAsNumber()/val);
+gdjs.Variable.prototype.div = function (val) {
+  this.setNumber(this.getAsNumber() / val);
 };
 
 /**
  * Concatenate the given string at the end of the variable value
- * @method
- * @param {string} str the string to append
+ * @param {string} str
+ */
+gdjs.Variable.prototype.concatenateString = function (str) {
+  this.setString(this.getAsString() + str);
+};
+
+/**
+ * @deprecated
+ * @private
+ */
+gdjs.Variable.prototype.concatenate = gdjs.Variable.prototype.concatenateString;
+
+/**
+ * Returns the variable at a specific index in the array.
+ * If it doesn't yet exists, create it.
+ * @param {number} index
+ * @returns {gdjs.Variable}
+ */
+gdjs.Variable.prototype.getAtIndex = function (index) {
+  if (this._childrenList[index] === undefined) {
+    this._type = 'array';
+    this._childrenList[index] = new gdjs.Variable();
+  }
+  return this._childrenList[index];
+};
+
+/**
+ * Removes a variable at a given index of the variable.
+ * @param {number} index
+ */
+gdjs.Variable.prototype.removeAtIndex = function (index) {
+  if (this._type === 'array') this._childrenList.splice(index, 1);
+};
+
+/**
+ * Pushes a variable onto the array.
+ * @param {gdjs.Variable} variable
  */
-gdjs.Variable.prototype.concatenate = function(str) {
-	this.setString(this.getAsString()+str);
+gdjs.Variable.prototype.push = function (variable) {
+  if (this._type === 'array' && variable instanceof gdjs.Variable)
```

Comment:
Why this `typeof`? Is the static typing not enough?

Reply:
Yes, I am concerned about users thinking this is accepting a javascript variable when using it from JS, like the push method of JS arrays.

Reply:
Might be worth outputing a warning in the console then?